routing/kbucket/table.go

```diff
@@ -48,11 +48,11 @@ func NewRoutingTable(bucketsize int, localID ID, latency time.Duration, m pstore
 // Update adds or moves the given peer to the front of its respective bucket
 // If a peer gets removed from a bucket, it is returned
 func (rt *RoutingTable) Update(p peer.ID) {
-	rt.tabLock.Lock()
-	defer rt.tabLock.Unlock()
 	peerID := ConvertPeerID(p)
 	cpl := commonPrefixLen(peerID, rt.local)
 
+	rt.tabLock.Lock()
+	defer rt.tabLock.Unlock()
```

Comment:
this no longer protects `rt.local`, which is fine. BUT it would be nice if the `RoutingTable` struct was more clear about what the mutex protects.

### now

```go
// RoutingTable defines the routing table.
type RoutingTable struct {

	// ID of the local peer
	local ID

	// Blanket lock, refine later for better performance
	tabLock sync.RWMutex

	// latency metrics
	metrics pstore.Metrics

	// Maximum acceptable latency for peers in this cluster
	maxLatency time.Duration

	// kBuckets define all the fingers to other nodes.
	Buckets    []*Bucket
	bucketsize int
}
```

### better

```go
// RoutingTable defines the routing table.
type RoutingTable struct {

	local      ID  // ID of the local peer
	bucketsize int

	// the following fields are all protected by tabLock
	tabLock    sync.RWMutex   // Blanket lock, refine later for better performance
	metrics    pstore.Metrics // latency metrics
	maxLatency time.Duration  // Maximum acceptable latency for peers in this cluster
	Buckets    []*Bucket      // kBuckets define all the fingers to other nodes.
}
```